integration-tests/tests/src/tests/queries.ts

```diff
@@ -139,9 +190,9 @@ const expectQueryResultValues = (
   queryResultPairs.forEach(([expectedResults, queryString, ...queryArgs]) => {
     let results = realm.objects<any>(objectSchema);
     results = results.filtered(queryString, ...queryArgs);
-    expect(expectedResults.length).equals(results.length);
-    expect(expectedResults).to.deep.equal(
-      results.map((el) => el[propertyToCompare]),
+    expect(results.length).to.equal(expectedResults.length);
```

Comment:
This uses the correct order of arguments, so that the text that is printed to console from the test is correct ("expected" vs "result")